### PR TITLE
[WIP] Adding handling for cryptography

### PR DESCRIPTION
**Pull request by masnes**

### CODE REVIEW

File: pa5-encfs.c [hunk 1]
```diff
@@ -495,21 +496,21 @@ int main(int argc, char *argv[])
...
-    else if (strcmp(argv[3], "-d") == 0 || strcmp(argv[3], "--decrypt") == 0)
-      decrypt = 1;
-    else  {
-      fprintf(stderr, "Error: 3rd argmument given does not specify encryption or decryption\n");
-      fprintf(stderr, "%s\n", usage);
-      return EXIT_FAILURE;
-    }
-  }
+	if (argc == 5) {  /* encrypt/decrypt and password */
+		if (strcmp(argv[3], "-e") == 0 || strcmp(argv[3], "--encrypt") == 0) {
```
Comment (ianks) [inline pa5-encfs.c:550]: I think we should be able to use [optparse](http://linux.die.net/man/3/getopt) here, yeah?

File: pa5-encfs.c [hunk 2]
```diff
@@ -495,21 +496,21 @@ int main(int argc, char *argv[])
...
+			encrypted = 1;
+			strncpy(password, argv[4], MAX_BUF_SIZE);
```
Comment (ianks) [inline pa5-encfs.c:557]: `strlcopy()` might be preferred here since it adds the terminating byte automatically.

File: pa5-encfs.c [hunk 3]
```diff
@@ -24,6 +24,7 @@
 
 #define FUSE_USE_VERSION 28
 #define HAVE_SETXATTR
+#define MAX_BUF_SIZE 400
```
Comment (ianks) [inline pa5-encfs.c:27]: Good call.

File: pa5-encfs.c [hunk 4]
```diff
@@ -495,21 +496,21 @@ int main(int argc, char *argv[])
...
+    } else {
+			fprintf(stderr, "Error: 3rd argmument present, but does not specify encryption\n");
+			fprintf(stderr, "%s\n", usage);
+			return EXIT_FAILURE;
+		}
+	}
 
 	argv[1] = argv[2];
 	argv[2] = NULL;
-	argc--;
```
Comment (ianks) [inline pa5-encfs.c:512]: We should probably be tidy here and NULLify `argv[3]` and `argv[4]`.

File: pa5-encfs.c (hunk 1 above)
Comment (masnes) [inline pa5-encfs.c:550]: Sounds good to me, although I don't think it's a priority

File: pa5-encfs.c (hunk 4 above)
Comment (masnes) [inline pa5-encfs.c:512]: done

File: pa5-encfs.c (hunk 2 above)
Comment (masnes) [inline pa5-encfs.c:557]: done

Comment (masnes) [inline pa5-encfs.c:557]: I'm actually doing an about-face on this because I just checked, and the VM doesn't have bsd functions.

Comment (ianks) [inline pa5-encfs.c:557]: Ah fair enough!

**Comment by ianks** [comment]
:+1:

**Pull request merged**